app/api/migrations/migrations/47-delete-superfluous-title-translations/index.js

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable no-await-in-loop */
+import { ObjectId } from 'mongodb';
+
+export default {
+  delta: 47,
+
+  name: 'delete-superfluous-title-translations',
+
+  description:
+    'The migration deletes the translations for the template &#x27;title&#x27; common property names that do not exist anymore (have been changed).',
+
+  async up(db) {
+    process.stdout.write(`${this.name}...\r\n`);
+
+    const translationCursor = await db.collection('translations').find({});
+    while (await translationCursor.hasNext()) {
+      const translation = await translationCursor.next();
+
+      await Promise.all(
+        translation.contexts.map(async context => {
+          if (context.type === 'Entity') {
+            const template = await db
+              .collection('templates')
+              .findOne({ _id: ObjectId(context.id) });
+
+            if (template) {
+              const templateLabelSet = new Set([
+                template.name,
+                'Title',
+                template.commonProperties.find(p => p.name === 'title').label,
+                ...template.properties.map(p => p.label),
+              ]);
+              context.values = context.values.filter(element => templateLabelSet.has(element.key));
+            }
+          }
+          return Promise.resolve();
```

Comment:
Is this return needed?

Reply:
It isn't. I thought Promise.all will complain, but it looks like it's more clever than I gave it credit for.

Reply:
It's a combination. Promise.all will accept non-promise values and treat them as if you returned Promise.resolve(value). But also, since you made your callback `async` it is always returning a promise. In your case, since you include no return value, it is returning a promise that resolves to an empty value, which is in fact the same you tried to achieve with your return :)